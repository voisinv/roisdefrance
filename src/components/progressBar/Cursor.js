// @flow
import React from 'react';

import { progressBarWidth } from '../../utils/progressBar';
import {getWindowWidth} from '../../utils/document';
const width = getWindowWidth();

export default (props: {x: number}) => {
  return (
    <g>
      <rect x={0} y={0} width={props.x} height={75}
            style={styles.noProgression}/>
      <rect x={props.x} y={0} width={progressBarWidth} height={75}
            style={styles.progression}/>
      <rect x={props.x + progressBarWidth} y={0}
            width={width - props.x - progressBarWidth}
            height={75} style={styles.noProgression}/>
    </g>);
}

const styles = {
  progression: {
    fill: 'none',
    stroke: 'black',
    strokeWidth: 2
  },
  noProgression: {
    fill: 'grey',
    opacity: 0.2
  }
};