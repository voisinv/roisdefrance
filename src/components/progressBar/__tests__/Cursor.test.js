import React from 'react';
import renderer from 'react-test-renderer';

import Cursor from '../Cursor';

describe('Cursor', () => {
  it('renders correctly', () => {
    const element = renderer.create(<Cursor x={10}></Cursor>).toJSON();
    expect(element).toMatchSnapshot();
  });
});