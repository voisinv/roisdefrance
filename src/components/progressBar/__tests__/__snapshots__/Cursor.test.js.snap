// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cursor renders correctly 1`] = `
<g>
  <rect
    height={75}
    style={
      Object {
        "fill": "grey",
        "opacity": 0.2,
      }
    }
    width={10}
    x={0}
    y={0}
  />
  <rect
    height={75}
    style={
      Object {
        "fill": "none",
        "stroke": "black",
        "strokeWidth": 2,
      }
    }
    width={40}
    x={10}
    y={0}
  />
  <rect
    height={75}
    style={
      Object {
        "fill": "grey",
        "opacity": 0.2,
      }
    }
    width={974}
    x={50}
    y={0}
  />
</g>
`;
